lib/src/sql/part.rs

```diff
@@ -89,6 +89,15 @@ impl From<&str> for Part {
 }
 
 impl Part {
+	/// Check if we require a writeable transaction
+	pub(crate) fn writeable(&self) -> bool {
+		match self {
+			Part::Where(v) => v.writeable(),
+			Part::Value(v) => v.writeable(),
+			Part::Method(_, v) => v.iter().any(Value::writeable),
+			_ => false,
+		}
+	}
```

Comment:
The criteria for checking the contents for write-ability seems to be whether a part contains one or more `Value`'s. Are we intentionally ignoring those that are within `Graph`?

(I still think this is worth merging without `Graph` `Value` support)

Reply:
Ah yeah that's a good point @finnbear. Although it is unlikely, we do need to check for values within a graph traversal query too yes. Separate PR!